### PR TITLE
Attempts to add timeout support for lua unit commands

### PR DESCRIPTION
**Pull request by tomjn**
e.g. something like this:

```lua
    Spring.GiveOrderToUnit(unitID,
      CMD.INSERT,
      {-1,CMD.ATTACK,''CMD.OPT_SHIFT'',''unitID2''},
      {"timeout" = 500 }
    );
```

Related to https://springrts.com/mantis/view.php?id=6128

### CODE REVIEW

**Comment by rtri** [comment]
That would work, but the (badly named) command "options" have special meaning. Timeout is a separate parameter.

**Comment by tomjn** [comment]
@rtri I appreciate the comment but you closed it before I could fix the PR!
No need for the trigger happy close, unless you're trying to put off
contributions?

Perhaps next time you should use the built in review functionality and
request changes? E.g. switching to an additional parameter because options
have special meaning?

**Comment by rtri** [comment]
You seem to have missed be999101, which I started before this PR arrived.

**Comment by tomjn** [comment]
@rtri **that's no excuse for rudeness**, I opened the PR to try and be helpful.

Next time:

 - If you already have a PR, link to it
 - Leave a review requesting changes

For all you know I may have come back and refactored it shortly after. If it had been a few days then sure, close it, as is common practice here, but instead you were hostile. Stop that.

**Comment by abma** [comment]
@tomjn:
next time: please make a pull request to the develop branch. we don't accept pull requests to the master branch!

**Comment by tomjn** [comment]
@abma I wasn't aware, perhaps a [pull request template](https://github.blog/2016-02-17-issue-and-pull-request-templates/) would help document this? That way it would be clear to anybody creating a pull request that they need to apply it to the `develop` branch not the `master` branch, even a prompt to link to any related Mantis tickets

---

As a general note, as a part of my job and general activities I regularly make and comment on pull requests on a wide range of projects on a daily basis, but this experience was poor, and below average, and entirely avoidable.

I don't know if the rude response from @rtri was because english isn't the primary language, or habit, but it's very easy to be polite and at the same time close responses. GitHub provides various processes to avoid these.

For example, the original comment doesn't explain why the PR was closed. If it had instead just been:

> Thanks, it's been fixed in [be99910](https://github.com/spring/spring/commit/be999101be20d10fe46ce6921d7ec7d88ccdece5)

That would have been perfectly acceptable on its own. The follow up comment was also catty and rude. Or a review left that was literally just the comment he left, requesting changes.

**Comment by abma** [comment]
@tomjn 
https://springrts.com/wiki/Development:Getting_Started#How_to_get_your_code_included
i presume you knew that. please stop complaining.

**Comment by rtri** [comment]
@tomjn 

 > the original comment doesn't explain why the PR was closed

When closing it I presumed you had taken at least a cursory glance at the log (or the mantis ticket) and would understand your PR had crossed incoming commits, which was clearly a mistake. Now kindly get off your lecturing high horse.

**Comment by tomjn** [comment]
>  i presume you knew that. please stop complaining.

No, I was not aware of that link, nor would anybody else who tried to create a pull request. At the very least link to it on the repos readme, if not a PR template so that it's shown on all new PR's

> When closing it I presumed you had taken at least a cursory glance at the log (or the mantis ticket) and would understand your PR had crossed incoming commits, which was clearly a mistake on my part. Now please get off your lecturing high horse.

I'd already started the pull request before the Mantis ticket had even been created. When you close a PR, explain why, a simple link to the commit will do.

**TLDR:** Don't just complain about a PR then close it without explanation, it's rude. And certainly don't argue that it wasn't rude, or be rude about it

I honestly don't see why there's such a fuss, but that either of you are still arguing implies the message was not received. Maybe take a page from Kloots book?
